### PR TITLE
GODRIVER-2248 Use mock deployment for slow BulkWrite update test

### PR DESCRIPTION
**Pull request by benjirewis**
GODRIVER-2248

`TestCollection/bulk_write/update_with_batches` was causing timeouts on most of our MacOS variants. This was because of the large (100050) amount of documents that needed to be deleted server-side after the test. As in GODRIVER-1858, I've used the mock deployment built into `mtest` to lower testing times (from about 1m to about .3s).

### CODE REVIEW

File: mongo/integration/collection_test.go [hunk 1]
```diff
@@ -1778,29 +1800,11 @@ func TestCollection(t *testing.T) {
...
-			// find the upserted documents and check their contents
-			id1, ok := res.UpsertedIDs[0]
+			// Check that IDs exist in result for upserted documents.
+			_, ok := res.UpsertedIDs[0]
 			assert.True(mt, ok, "expected id at key 0")
-			id2, ok := res.UpsertedIDs[int64(numModels-1)]
+			_, ok = res.UpsertedIDs[int64(numModels-1)]
 			assert.True(mt, ok, "expected id at key %v", numModels-1)
-
-			doc, err := mt.Coll.FindOne(mtest.Background, bson.D{{"_id", id1}}).DecodeBytes()
```
Comment (benjirewis) [inline mongo/integration/collection_test.go:1787]: I've removed the calls to `FindOne` to verify that the upserted documents exist. I _could_ mock responses again here, but that doesn't seem to be testing the functionality of a bulk write update with batches.

**Review by benjirewis — COMMENTED** [review]
I've linked a number of MacOS variants to this PR's CI so we can verify that timeouts are mostly gone.

File: mongo/integration/collection_test.go (hunk 1 above)
Comment (kevinAlbs) [reply mongo/integration/collection_test.go:1787]: I agree the `FindOne` does not seem necessary. This test was motivated by a bug in merging bulk update results: GODRIVER-1715

**Review by kevinAlbs — APPROVED** [review]
LGTM

**Pull request merged**